mingw-w64-git/git-wrapper.c

```diff
@@ -120,8 +120,13 @@ static void setup_environment(LPWSTR top_level_path, int full_path)
 	int len;
 
 	/* Set MSYSTEM */
+	#if defined(IS_ARM64) && IS_ARM64 == true
```

Comment:
It might be more robust to make this dependent on, say, the presence of a sub-directory. For example, we could consider putting the ARM64 binaries into the `<MSYS2-pseudo-root>/arm64/bin/` directory. Then, testing for `arm64/bin` could serve as condition, and we wouldn't have a compile-time switch.

What do you think?

Reply:
Or even better, we could determine whether we're running on an ARM64 processor via [`GetNativeSystemInfo()`](https://docs.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getnativesysteminfo) and [testing `result.wProcessorArchitecture == PROCESSOR_ARCHITECTURE_ARM64`](https://docs.microsoft.com/en-us/windows/win32/api/sysinfoapi/ns-sysinfoapi-system_info#members). If that is the case, we can always add the directory to the `PATH`, even if the directory does not exist (because it does not hurt in that case, it's just ignored).

Reply:
@dscho I like this idea. Just spent 2 hours trying to get it to work (I'm new to C(++)), turns out that the generated i686 executable returns `PROCESSOR_ARCHITECTURE_INTEL` instead of `PROCESSOR_ARCHITECTURE_ARM64` due to the 32-bit emulation 🤦🏼 

The generated i686 executable:

![image](https://user-images.githubusercontent.com/17739158/103823725-ed243900-5072-11eb-975f-57813ee5b753.png)

![image](https://user-images.githubusercontent.com/17739158/103823877-32486b00-5073-11eb-85e2-7123e9ba7e8a.png)

When I create an example C++ file and compile it for ARM64 natively, it does report the correct arch:

![image](https://user-images.githubusercontent.com/17739158/103823834-252b7c00-5073-11eb-8721-9de287eb2810.png)

So shall I proceed with checking for the `arm64/bin` folder at runtime, or do you have any other suggestions?

Reply:
Would `PathIsDirectoryA` from `shlwapi.h` be worth checking out you think?

Reply:
> turns out that the generated i686 executable returns `PROCESSOR_ARCHITECTURE_INTEL` instead of `PROCESSOR_ARCHITECTURE_ARM64` due to the 32-bit emulation 🤦🏼

Is this with `GetNativeSystemInfo()` or with `GetSystemInfo()`?

Reply:
@dscho  Both. Should've mentioned that, sorry.

Running x86 executable on an AMD64 machine:

![image](https://user-images.githubusercontent.com/17739158/104002583-92d5c600-51a1-11eb-9002-96a70e5e9df0.png)

Running x86 executable on an ARM64 machine (emulation):

![arm64](https://user-images.githubusercontent.com/17739158/104002678-af71fe00-51a1-11eb-8e98-9648a409a531.png)

Reply:
Hmpf. Next guess... [`IsWow64Process2()`](https://docs.microsoft.com/en-gb/windows/win32/api/wow64apiset/nf-wow64apiset-iswow64process2)?